src/minmax.rs

```diff
@@ -0,0 +1,90 @@
+use rustc::lint::{Context, LintPass, LintArray};
+use rustc_front::hir::*;
+use syntax::codemap::Spanned;
+use syntax::ptr::P;
+use std::cmp::PartialOrd;
+use std::cmp::Ordering::*;
+
+use consts::{Constant, constant};
+use utils::{match_path, span_lint};
+use self::MinMax::{Min, Max};
+
+declare_lint!(pub MIN_MAX, Deny,
```

Comment:
And on the topic of complicated code/reasoning: Are you talking about clippy's code or code we run cilppy against?

Because in the latter case, I'd argue that it won't matter. Regardless of simple or complex code, if the lint catches it, it's wrong. Plain and simple (well, unless I have made a mistake in the code or the math that results in a false positive). This is the same reasoning with `cmp_nan`, `zero_width_space`, `bad_bit_mask` and `min_max`.

Otherwise we should add more tests, and run clippy against more code out in the wild. This is why I'm taking the time to review the result manually. I'm not only checking the code, but also looking for false positives.